File: Java/Livro.java
public class Livro {


    private String titulo;
    private String autor;
    private int anoPublicacao;
    private int numPag;
    private double peso;
    private double volume;

    public Livro(String titulo,String autor,int anoPublicacao,int numPag,double peso,double volume) {

        this.titulo = titulo;
        this.autor = autor;
        this.anoPublicacao = anoPublicacao;
        this.numPag = numPag;
        this.peso = peso;
        this.volume = volume;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getAutor() {
        return autor;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public int getAnoPublicacao() {
        return anoPublicacao;
    }

    public void setAnoPublicacao(int anoPublicacao) {
        this.anoPublicacao = anoPublicacao;
    }

    public int getNumPag() {
        return numPag;
    }

    public void setNumPag(int numPag) {
        this.numPag = numPag;
    }

    public double getPeso() {
        return peso;
    }

    public void setPeso(double peso) {
        this.peso = peso;
    }

    public double getVolume() {
        return volume;
    }

    public void setVolume(double volume) {
        this.volume = volume;
    }

    public void marcarPagina( int numero) {

        if (numero > 0 && numero <= numPag) {
            System.out.println("Livro marcado na página: " + numero);
        } else {
            System.out.println("Página inválida!");
        }
    }
}

File: Python/Livro.py
class Livro:
    def __init__(self, titulo, autor, anoDePublicacao, numeroDePaginas, peso, volume):
        self.__titulo = titulo
        self.__autor = autor
        self.__anoDePublicacao = anoDePublicacao
        self.__numeroDePaginas = numeroDePaginas
        self.__peso = peso
        self.__volume = volume

    def getTitulo(self):
        return self.__titulo

    def getAutor(self):
        return self.__autor

    def getAnoDePublicacao(self):
        return self.__anoDePublicacao

    def getNumeroDePaginas(self):
        return self.__numeroDePaginas

    def getPeso(self):
        return self.__peso

    def getVolume(self):
        return self.__volume

    def setTitulo(self, titulo):
        self.__titulo = titulo

    def setAutor(self, autor):
        self.__autor = autor

    def setAnoDePublicacao(self, anoDePublicacao):
        self.__anoDePublicacao = anoDePublicacao

    def setNumeroDePaginas(self, numeroDePaginas):
        self.__numeroDePaginas = numeroDePaginas

    def setPeso(self, peso):
        self.__peso = peso

    def setVolume(self, volume):
        self.__volume = volume

    def marcarPagina(self, numero):
        if numero >= 1 and numero <= self.__numeroDePaginas:
            print("Livro marcado na página " + str(numero))
        else:
            print("Página inválida!")
